examples/cpp/hello_world/HelloWorldSubscriber.cpp

```diff
@@ -19,99 +19,70 @@
+#include <fastdds/dds/subscriber/qos/SubscriberQos.hpp>
 #include <fastdds/dds/subscriber/SampleInfo.hpp>
 #include <fastdds/dds/subscriber/Subscriber.hpp>
 
 using namespace eprosima::fastdds::dds;
 
+std::atomic<bool> HelloWorldSubscriber::stop_(false);
+std::mutex HelloWorldSubscriber::terminate_cv_mtx_;
+std::condition_variable HelloWorldSubscriber::terminate_cv_;
+
 HelloWorldSubscriber::HelloWorldSubscriber()
     : participant_(nullptr)
     , subscriber_(nullptr)
     , topic_(nullptr)
     , reader_(nullptr)
     , type_(new HelloWorldPubSubType())
 {
-}
-
-bool HelloWorldSubscriber::init(
-        bool use_env)
-{
-    DomainParticipantQos pqos = PARTICIPANT_QOS_DEFAULT;
-    pqos.name("Participant_sub");
+    // Create the participant
     auto factory = DomainParticipantFactory::get_instance();
-
-    if (use_env)
-    {
-        factory->load_profiles();
-        factory->get_default_participant_qos(pqos);
-    }
-
-    participant_ = factory->create_participant(0, pqos);
-
+    participant_ = factory->create_participant_with_default_profile();
     if (participant_ == nullptr)
     {
-        return false;
+        throw std::runtime_error("Participant initialization failed");
     }
 
-    //REGISTER THE TYPE
+    // Register the type
     type_.register_type(participant_);
 
-    //CREATE THE SUBSCRIBER
-    SubscriberQos sqos = SUBSCRIBER_QOS_DEFAULT;
-
-    if (use_env)
-    {
-        participant_->get_default_subscriber_qos(sqos);
-    }
-
-    subscriber_ = participant_->create_subscriber(sqos, nullptr);
-
+    // Create the subscriber
+    SubscriberQos sub_qos = SUBSCRIBER_QOS_DEFAULT;
+    participant_->get_default_subscriber_qos(sub_qos);
+    subscriber_ = participant_->create_subscriber(sub_qos, nullptr);
     if (subscriber_ == nullptr)
     {
-        return false;
-    }
-
-    //CREATE THE TOPIC
-    TopicQos tqos = TOPIC_QOS_DEFAULT;
-
-    if (use_env)
-    {
-        participant_->get_default_topic_qos(tqos);
+        throw std::runtime_error("Subscriber initialization failed");
     }
 
-    topic_ = participant_->create_topic(
-        "HelloWorldTopic",
-        "HelloWorld",
-        tqos);
-
+    // Create the topic
+    TopicQos topic_qos = TOPIC_QOS_DEFAULT;
+    participant_->get_default_topic_qos(topic_qos);
+    topic_ = participant_->create_topic("Hello_world_topic", "HelloWorld", topic_qos);
     if (topic_ == nullptr)
     {
-        return false;
-    }
-
-    // CREATE THE READER
-    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
-    rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
-
-    if (use_env)
-    {
-        subscriber_->get_default_datareader_qos(rqos);
+        throw std::runtime_error("Topic initialization failed");
     }
 
-    reader_ = subscriber_->create_datareader(topic_, rqos, &listener_);
-
+    // Create the reader
+    DataReaderQos reader_qos = DATAREADER_QOS_DEFAULT;
+    reader_qos.reliability().kind = RELIABLE_RELIABILITY_QOS;
```

Comment:
I guess we should document this

Reply:
As discussed internally, this configuration would be part of the XML files